### PR TITLE
Feature/targets endpoint

### PR DESCRIPTION
**Pull request by agnessa**
For the targets endpoint the database part is needed. Therefore:
- `bundle exec rake db:create db:migrate db:seed`

This creates three tables:
- `reports` - a report will be linked to a country / user at a later stage when accounts are in place; right now a default report is created by seeds and instantiated with defaults for testing purposes
- `categories` - categories present in this table (linked by slug) are active for a report
- `targets` - for now only generic targets, no support for user-defined ones

Adds an endpoint to fetch targets for category:
`GET /api/v1/sections/planning/categories/ndc_targets/targets?year=2018`

Documentation of the endpoint in docs.

### CODE REVIEW

File: app/services/merge_static_and_dynamic_targets.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def call
+    @report_targets.map do |report_target|
+      match = @static_targets.detect do |static_target|
+        report_target.slug == static_target.slug
+      end
+      target_hash = match&.to_hash || {} # TODO: custom targets
+      target_hash[:year] = report_target.year
+      target_hash[:reported_percentage] = 0 # TODO:
```
Comment (weberjavi) [inline app/services/merge_static_and_dynamic_targets.rb]: Not sure if the `reported_percentage` attribute is needed on targets under the `Planning` section.  
Beyond that everything is working fine :-)

**Review by weberjavi — APPROVED** [review]
From a non Rails savvy perspective every thing looks good to me 😀 
Feel free to merge the PR.

**Comment by Bluesmile82** [comment]
The endpoints:
localhost:3000/api/v1/sections?includes[]=categories&includes[]=targets
localhost:3000/api/v1/sections/planning/categories?includes[]=targets
Doesn't seem to show the targets even if there are some for ndc_targets. Could you please check that?

**Review by Bluesmile82 — CHANGES_REQUESTED** [review]
Check my comment but great work!

**Comment by agnessa** [comment]
@Bluesmile82 thank you for your comment, I have a feeling that this might happen when some temporary initialisation happens in the wrong order, e.g. when you try the endpoint before running the seeds, and you end up with more than one report object where one is initialised and the other empty. Could you try `bundle exec rake db:seeds` again - I guess it will fix it in your db. The problem here is we don't have yet the user - report relationships in place so there's some temporary mock ups in place that will be deleted as we keep integrating, so probably it's not bullet proof at it is.

**Review by Bluesmile82 — APPROVED** [review]

**Pull request merged**